### PR TITLE
Fix addon-prometheus by updating some version numbers and file permissions

### PR DESCRIPTION
**Pull request by 42Network**
# Proposed Changes

> Bump prometheus to 2.37.0
> Downgrade wheels to alpine-3.14
> Eliminate /data/prometheus permission issue by adding chmod -R to prometheus.sh script
> Correct location of libraries

## Related Issues

> ([Github link][autolink-references] to related issues or pull requests)

[autolink-references]: https://github.com/hassio-addons/addon-prometheus/issues/48#issue-1271152226

### CODE REVIEW

File: prometheus/Dockerfile [hunk 1]
```diff
@@ -7,7 +7,7 @@ SHELL ["/bin/bash", "-o", "pipefail", "-c"]
 
 # Setup base system
 ARG BUILD_ARCH=amd64
-ENV PROMETHEUS_VERSION=2.36.1
+ENV PROMETHEUS_VERSION=2.37.0
```
Comment (frenck) [inline prometheus/Dockerfile:10]: This should be in a separate PR.

File: prometheus/Dockerfile [hunk 2]
```diff
@@ -43,7 +43,7 @@ RUN \
     && pip3 install \
         --no-cache-dir \
         --prefer-binary \
-        --find-links "https://wheels.home-assistant.io/alpine-3.15/${BUILD_ARCH}/" \
+        --find-links "https://wheels.home-assistant.io/alpine-3.14/${BUILD_ARCH}/" \
```
Comment (frenck) [inline prometheus/Dockerfile:46]: This is incorrect, as a matter of fact, the image is not 3.14, please remove this change.

File: prometheus/rootfs/etc/cont-init.d/prometheus.sh [hunk 3]
```diff
@@ -6,5 +6,6 @@
 echo "${SUPERVISOR_TOKEN}" > '/run/home-assistant.token'
 
 mkdir -p /data/prometheus
+chown -R prometheus:prometheus /data/prometheus
 find /data/prometheus -not -perm 0644 -type f -exec chmod 0660 {} \;
 find /data/prometheus -not -perm 0755 -type d -exec chmod 0770 {} \;
```
Comment (frenck) [inline prometheus/rootfs/etc/cont-init.d/prometheus.sh:11]: We should migrate the add-on to run as root instead (as per the general design of Home Assistant).

**Review by frenck — CHANGES_REQUESTED** [review]

File: prometheus/Dockerfile (hunk 2 above)
Comment (42Network) [reply prometheus/Dockerfile:46]: https://wheels.home-assistant.io/alpine-3.15/amd64/ gives a 404 error, whereas alpine-3.14 works.
image

File: prometheus/Dockerfile (hunk 1 above)
Comment (42Network) [reply prometheus/Dockerfile:10]: Ok, I will resubmit at the 2.36.1 version. But since 2.37.0 is the LTS version of Prometheus it would be good to use that one.

File: prometheus/rootfs/etc/cont-init.d/prometheus.sh (hunk 3 above)
Comment (42Network) [reply prometheus/rootfs/etc/cont-init.d/prometheus.sh:11]: Yes, that would probably eliminate the need for checking the ownership and permissions of the /data/prometheus/* files. Did you do the original work on this addon? I may take a stab at converting it to run as root, but not sure when I can get to it.

Thanks for working with me on my very first HA PR.

-Nathan

File: prometheus/Dockerfile (hunk 2 above)
Comment (frenck) [reply prometheus/Dockerfile:46]: Correct :)
That is fine for now.

This code will fully change in the new base images I'm rolling out anyways

Comment (42Network) [reply prometheus/Dockerfile:46]: Are you saying that is fine to leave it as 3.14 for now, or that it is fine for the find-links to fail when the version is 3.15?

**Comment by mcarbonneaux** [comment]
this fix can be commited to fix the "open /data/prometheus/queries.active: permission denied" error ?

**Comment by frenck** [comment]
Thanks for the PRs!

However, I've decided to no longer pursue the further development of this add-on. Up until now it has only been in edge, and was never released stable.

Feel free to pick up the sources to publish your own.

../Frenck
